### PR TITLE
[TT-10654/TT-10702] Go 1.21

### PR DESCRIPTION
**Pull request by jeffy-mathew**
## Description

Upgrade go version to 1.21
## Related Issue
Parent issue https://tyktech.atlassian.net/browse/TT-10654
Sub task https://tyktech.atlassian.net/browse/TT-10702

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

### CODE REVIEW

**Comment by buger** [comment]
**API tests result - mongo44-sha256 env: skipped** :no_entry_sign: 
Branch used: refs/pull/5830/merge
Commit: 8a2e9917a071017e03866f65c5a2206c965243c8 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7082883025)

**Comment by buger** [comment]
**API tests result - mongo44-murmur64 env: skipped** :no_entry_sign: 
Branch used: refs/pull/5830/merge
Commit: 8a2e9917a071017e03866f65c5a2206c965243c8 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7082883025)

**Comment by buger** [comment]
**API tests result - postgres15-sha256 env: skipped** :no_entry_sign: 
Branch used: refs/pull/5830/merge
Commit: 8a2e9917a071017e03866f65c5a2206c965243c8 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7082883025)

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5830/merge
Commit:  
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7082834790)

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5830/merge
Commit: 8a2e9917a071017e03866f65c5a2206c965243c8 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7082882983)

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5830/merge
Commit: 44a6ba6f693f88e5360f027a029a67f131fbec57 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7083180552)

File: ci/Dockerfile.std [hunk 1]
```diff
@@ -23,7 +25,7 @@ RUN rm -rf /root/.cache \
     && rm -rf /usr/include/* \
     && find /usr/lib -type f -name '*.a' -o -name '*.o' -delete
 # Print included pip/python versions
-RUN pip3 --version && python3 --version
+RUN python3 --version
```
Comment (titpetric) [inline ci/Dockerfile.std:28]: This looks off, pip check removed

**Review by titpetric — COMMENTED** [review]
There's uhm some changes that don't make sense for just the Go upgrade, do we need to rebase this? Didn't expect code change in internal/crypto, tests, plugin-aliasing seems out of date (moved those plugins into a testdata/) ....

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5830/merge
Commit: abf9c7d7af8b5836de10f522963f87597817abe2 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7084583073)

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5830/merge
Commit: 3ec0ac1ea45cd9ca4558289633a282b56d937437 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7084818926)

**Review by titpetric — APPROVED** [review]

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5830/merge
Commit: 7eb2555730c365f66d742b3cf5f1225e6624e3bf 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7085074821)

**Pull request merged**